Logic/data_processing.cpp

```diff
@@ -1,46 +1,107 @@
+#include "data_processing.hpp"
+#include "storage_structures.hpp"
 #include <cstdint>
 #include <iostream>
 #include <list>
+#include <map>
 #include <memory>
+#include <queue>
 #include <string>
 #include <thread>
 #include <unordered_map>
 #include <vector>
-#include "storage_structures.hpp"
-#include "data_processing.hpp"
-
-// #include "concurrentqueue.h"
 
 namespace p2p {
 
-// struct DataReceiver {
-//   void receive() {}
-// };
+void Analysis::analyze(Chains &chains, Orders &orders,
+                       MarketRates &market_rates) {
+  Orders orders_for_sell;
+  Orders orders_for_buy;
+  for (Order order : orders.list) {
+    if (order.type == "sell") {
+      orders_for_sell.list.push_back(order);
+    } else if (order.type == "buy") {
+      orders_for_buy.list.push_back(order);
+    }
+  }
+
+  if (ORDERS_FOR_BUY_LOGS_ON) {
+    std::cout << "orders_for_buy size: " << orders_for_buy.list.size()
+              << std::endl;
+    for (int i = 0; i < 5; i++) {
+      std::cout << i << ". Покупаем: " << orders_for_buy.list[i].coin2 << " за "
+                << orders_for_buy.list[i].coin1 << " по цене "
+                << orders_for_buy.list[i].exchange_rate << std::endl;
+    }
+  }
+
+  if (ORDERS_FOR_SELL_LOGS_ON) {
+    std::cout << "orders_for_sell size: " << orders_for_sell.list.size()
+              << std::endl;
+    for (int i = 0; i < orders_for_sell.list.size(); i++) {
+      std::cout << i << ". Продаём: " << orders_for_sell.list[i].coin2 << " за "
+                << orders_for_sell.list[i].coin1 << " по цене "
+                << orders_for_sell.list[i].exchange_rate << std::endl;
+    }
+  }
 
-void Analysis::analyze(Chains &chains, Orders &orders_for_buy, Orders &orders_for_sell,
-              MarketRates &market_rates) {
   std::unordered_map<std::string, std::vector<Order>> sell_by_coin;
   for (Order sell : orders_for_sell.list) {
-    sell_by_coin[sell.coin1].push_back(sell);
+    sell_by_coin[sell.coin2].push_back(sell);
+  }
+
+  if (SELL_BY_COIN_LOGS_ON) {
+    std::cout << "sell_by_coin size: " << sell_by_coin.size() << std::endl;
+    std::cout << "sell_by_coin:" << std::endl;
+    for (auto it = sell_by_coin.begin(); it != sell_by_coin.end(); ++it) {
+      std::cout << it->first << std::endl;
+    }
+  }
+
+  if (MARKET_RATES_LOGS_ON) {
+    std::cout << "market_rates:" << std::endl;
+    for (auto it = market_rates.list.begin(); it != market_rates.list.end();
+         ++it) {
+      for (auto it2 = it->second.begin(); it2 != it->second.end(); ++it2) {
+        std::cout << it->first << " " << it2->first << " " << it2->second
+                  << std::endl;
+      }
+    }
   }
+
+  std::unordered_map<std::string, Chain> chains_map;
   for (Order buy : orders_for_buy.list) {
+    long double current_sum_1 = 1000;
```

Comment:
nit) naming of the variables could be better `sum_1,sum_2,sum_3`, Also it's not very clear what 1000/100 constants do here